persist/fs/commitlog/writer.go

```diff
@@ -154,19 +157,38 @@ func (w *writer) isOpen() bool {
 	return w.chunkWriter.fd != nil
 }
 
+func (w *writer) logEntryIndex(seriesUniqueIndex uint64) uint64 {
+	// Special case the zeroth value as we compare the value
+	// at the entry of the seen cache with the index being put
+	// at the location - since all entries in the seen cache when
+	// not set are zero the value being used to compare whether
+	// this result has been seen or not cannot be zero itself
+	return seriesUniqueIndex + 1
+}
+
 func (w *writer) Write(
 	series Series,
 	datapoint ts.Datapoint,
 	unit xtime.Unit,
 	annotation ts.Annotation,
 ) error {
+	logEntryIndex := w.logEntryIndex(series.UniqueIndex)
```

Comment:
That would only redistribute the layout of ID -> index in the seen cache, which for the cost of a hash doesn't seem worth it.  I hashed 100 million IDs in a row and it took a long time vs just doing a modulus and an array lookup/insertion was hundreds of milliseconds, i.e. hash approach took 10s for 100 million and modulus lookup/insertion was 400ms.